cdap-formats/src/main/java/io/cdap/cdap/format/RecordPutTransformer.java

```diff
@@ -96,34 +96,41 @@ private void setField(Put put, Schema.Field field, Object val) {
 
     Schema.Type type = validateAndGetType(field);
 
-    switch (type) {
-      case BOOLEAN:
-        put.add(field.getName(), (Boolean) val);
-        break;
-      case INT:
-        put.add(field.getName(), (Integer) val);
-        break;
-      case LONG:
-        put.add(field.getName(), (Long) val);
-        break;
-      case FLOAT:
-        put.add(field.getName(), (Float) val);
-        break;
-      case DOUBLE:
-        put.add(field.getName(), (Double) val);
-        break;
-      case BYTES:
-        if (val instanceof ByteBuffer) {
-          put.add(field.getName(), Bytes.toBytes((ByteBuffer) val));
-        } else {
-          put.add(field.getName(), (byte[]) val);
-        }
-        break;
-      case STRING:
-        put.add(field.getName(), (String) val);
-        break;
-      default:
-        throw new IllegalArgumentException("Field " + field.getName() + " is of unsupported type " + type);
+    try {
+      switch (type) {
+        case BOOLEAN:
+          put.add(field.getName(), (Boolean) val);
+          break;
+        case INT:
```

Comment:
For numeric type, shouldn't we allow upcasting (e.g. int value is allowed for long type)?

Reply:
we could, but I think this would be the wrong place to do it. If we allow it, I think it should be when building a StructuredRecord (such that setting an int for a long field converts the int to a long), so that anything that uses StructuredRecord doesn't have to do similar upcasting.